langchain/llms/azureml_endpoint.py

```diff
@@ -0,0 +1,238 @@
+"""Wrapper around AzureML Managed Online Endpoint API."""
+import json
+import urllib.request
+from abc import abstractmethod
+from typing import Any, Dict, Generic, List, Mapping, Optional, TypeVar, Union
+
+from pydantic import BaseModel, validator
+
+from langchain.callbacks.manager import CallbackManagerForLLMRun
+from langchain.llms.base import LLM
+from langchain.utils import get_from_dict_or_env
+
+INPUT_TYPE = TypeVar("INPUT_TYPE", bound=Union[str, List[str]])
+OUTPUT_TYPE = TypeVar("OUTPUT_TYPE", bound=Union[str, List[List[float]]])
+
+
+class AzureMLEndpointClient(object):
+    """Wrapper around AzureML Managed Online Endpoint Client."""
+
+    def __init__(
+        self, endpoint_url: str, endpoint_api_key: str, deployment_name: str
+    ) -> None:
+        """Initialize the class."""
+        self.endpoint_url = endpoint_url
+        self.endpoint_api_key = endpoint_api_key
+        self.deployment_name = deployment_name
+
+    def call(self, body: bytes) -> Union[Dict, str]:
+        """call."""
+
+        url = self.endpoint_url
+        # Replace this with the primary/secondary key or AMLToken for the endpoint
+        api_key = self.endpoint_api_key
+        if not api_key:
+            raise Exception("A key should be provided to invoke the endpoint")
+
+        # The azureml-model-deployment header will force the request to go to a
+        # specific deployment. Remove this header to have the request observe the
+        # endpoint traffic rules.
+
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": ("Bearer " + api_key),
+            "azureml-model-deployment": self.deployment_name,
+        }
+
+        req = urllib.request.Request(url, body, headers)
+        try:
+            response = urllib.request.urlopen(req, timeout=50)
+            result = response.read()
+        except urllib.error.HTTPError as error:
```

Comment:
i think we probably want to let users handle errors themselves instead of forcing something to be returned. if user did want to customize error handling current approach would make it hard/brittle, they'd have to parse and inspect outputs

Reply:
Makes sense 👍